pkg/controller/managedresources/controller.go

```diff
@@ -336,19 +341,29 @@ func (r *Reconciler) applyNewResources(newResourcesObjects []object, labelsToInj
 		}
 	)
 
+	// get all HPA and HVPA targetRefs to check if we should prevent overwriting replicas and/or resource requirements
```

Comment:
Maybe either here or below in the function's doc string - can you add a small comment why VPA is not checked/needed? :)

Reply:
Done for both here and below.